src/applications/letters/components/DownloadLetterLink.jsx

```diff
@@ -39,7 +39,6 @@ export class DownloadLetterLink extends React.Component {
         break;
       case DOWNLOAD_STATUSES.success:
         buttonText = 'Download letter';
-        buttonDisabled = false;
```

Comment:
I wonder if we might want to set this to `buttonDisabled = null` or `undefined` instead - not sure if we can go from a true state to a "false" state in this flow, but for safety

Reply:
I'm with you here. I did some more digging and the best corollary was this entry on Web.dev: https://web.dev/learn/javascript/data-types/null-undefined#comparison

> Although `undefined` and `null` have some functional overlap, they have different purposes. In the strictest sense, `null` represents a value intentionally defined as "blank," and `undefined` represents a lack of any assigned value.

Applying that over the HTML 5 spec on Boolean attributes "The values "true" and "false" are not allowed on boolean attributes. To represent a false value, the attribute has to be omitted altogether." it feels like `undefined` is a better choice.

I'll update this in the morning in functional code and unit tests. Thanks Stacy!

Reply:
I had to do a quick refresher to ensure `let` was `undefined` if declared with no value. MDN to the rescue! https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let

Reply:
While this is true, I was thinking there might be a scenario where the props might change and the component might re-render - but I suppose with the way it's set up as a classic style React class component, it will re-instantiate the whole thing, so my concern is invalid in this context. 😄

Reply:
Looking through this a second time, I missed what you were saying initially @dysbo. I went back and added the explicit `buttonDisabled = undefined;` to the `success | failure | default` cases respectively. All unit tests passed and quick manual retest was good too.